packages/flutter/lib/src/widgets/tap_outside_configuration.dart

```diff
@@ -0,0 +1,139 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' as ui hide TextStyle;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+
+import 'focus_manager.dart';
+import 'framework.dart';
+
+/// Describes how [EditableText] widgets should behave.
+///
+/// {@template flutter.widgets.TapOutsideBehavior}
+/// Used by [TapOutsideConfiguration] to configure the [EditableText] widgets in a
+/// subtree.
+///
+/// This class can be extended to further customize a [TapOutsideBehavior] for a
+/// subtree. For example, overriding [TapOutsideBehavior.defaultOnTapOutside] sets the
+/// default [EditableText.onTapOutside] for [EditableText]s that inherit this [TapOutsideConfiguration].
+/// {@endtemplate}
+///
+/// See also:
+///
+///   * [TapOutsideConfiguration], the inherited widget that controls how
+///     [EditableText] widgets behave in a subtree.
+@immutable
+class TapOutsideBehavior {
```

Comment:
Hey @gspencergoog. Thanks for the answer. 
I was thinking about more generic solution. I tried to implement it on the `TapRegion` level and pass `groupId` as parameter to `defaultOnTapOutside` so you could implement different behaviour for different groups. The problem with the more generic solution is that you may need more then just a `PointerDownEvent event` that `onTapOutside` from `TapRegion` gives you. For example, to implement default behavior of `EditableText.onTapOutside` you need also `FocusNode` and probably this will be the problem for other cases. I haven't found a solution to this. If you know any let me know and I will modify PR. Otherwise I will try to adjust naming. Is `EditTextTapOutsideConfiguration` ok?